dmutils/email/dm_notify.py

```diff
@@ -66,54 +80,66 @@ def has_been_sent(self, reference):
         return hash_string(details_string)
 
-    @staticmethod
-    def get_error_message(email_address, error):
-        """Format a logical error message from the error response."""
-        messages = []
-        message_prefix = u'Error sending message to {email_address}: '.format(email_address=email_address)
-        message_string = u'{status_code} {error}: {message}'
-
-        for message in error.message:
-            format_kwargs = {
-                'status_code': error.status_code,
-                'error': message['error'],
-                'message': message['message'],
-            }
-            messages.append(message_string.format(**format_kwargs))
-        return message_prefix + u', '.join(messages)
-
-    def send_email(self, email_address, template_id, personalisation=None, allow_resend=True, reference=None):
+    def _log_email_error_message(self, to_email_address, template_name, reference, error):
+        """Format a logical error message from the error response and send it to the logger"""
+
+        error_messages = []
+        for error_message in error.message:
+            error_messages.append(
+                "{status_code} {error}: {message}".format(
+                    status_code=error.status_code,
+                    error=error_message["error"],
+                    message=error_message["message"],
+                )
+            )
+
+        self.logger.error(
+            "Error sending email: {error_messages}",
+            extra={
+                "client": self.__class__,
+                "reference": reference,
+                "template_name": template_name,
+                "to_email_address": hash_string(to_email_address),
+                "error_messages": error_messages,
+            },
+        )
+
+    def send_email(self, to_email_address, template_name, personalisation=None, allow_resend=True, reference=None):
         """
         Method to send an email using the Notify api.
 
-        :param email_address: String email address for recipient
-        :param template_id: Template accessible on the Notify account whose  api_key you instantiated the class with
+        :param to_email_address: String email address for recipient
+        :param template_name: Template accessible on the Notify account whose api_key you instantiated the class with.
+                              Can either be a UUID or a key to the current_app.config["NOTIFY_TEMPLATES"] dictionary.
         :param personalisation: The template variables, dict
         :param allow_resend: if False instantiate the delivered reference cache and ensure we are not sending duplicates
         :return: response from the api. For more information see https://github.com/alphagov/notifications-python-client
         """
-        reference = reference or self.get_reference(email_address, template_id, personalisation)
+        template_id = self.templates.get(template_name, template_name)
```

Comment:
Hmmm I'm not sure about this conflation of `template_id` and `template_name`. If we want to be able to pass either to this method, we should probably have separate keyword args for them. I can just see us getting confused when we come to use it in one of the apps.